underfs/s3a/src/main/java/alluxio/underfs/s3a/S3AUnderFileSystem.java

```diff
@@ -410,6 +440,51 @@ public ObjectListingChunk getNextChunk() throws IOException {
     }
   }
 
+  /**
+   * Wrapper over S3 {@link ListObjectsRequest}.
+   */
+  private final class S3AObjectListingChunkV1 implements ObjectListingChunk {
+    final ListObjectsRequest mRequest;
+    final ObjectListing mResult;
+
+    S3AObjectListingChunkV1(ListObjectsRequest request, ObjectListing result)
```

Comment:
If `result` is not supposed to be `null`, we can use `Preconditions.checkNotNull(result, "result")` to verify.